### PR TITLE
Add yeoman generator for component

### LINKED ISSUE #119: Make it easier to create a new component
@rese had trouble creating a new component. Styleguidist only recognizes class components, not function components. Also copying an existing folder and renaming things is complex and error prone.

Maybe we can provide a yeoman generator?

### PR DESCRIPTION
**Pull request by unindented**
Copy-pasting and renaming is time consuming and error prone.

yo mov

Fixes #119

### CODE REVIEW

**Comment by unindented** [comment]
@swese44 @rese please review.

**Comment by rese** [comment]
Woohoo! Adding this template generating stuff will not only make things Rese-proof but also make development faster.

Thanks for doing this!

File: config/yo/generator-component/app/index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      {
+        type: 'input',
+        name: 'name',
+        message: 'What\'s the name of your new component? (e.g. MyNewComponent)',
+      },
+    ];
+
+    return this.prompt(prompts).then((props) => {
+      const { name } = props;
+      const className = `y-${name.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase()}`;
```
Comment (rese) [inline config/yo/generator-component/app/index.js]: It adds creates a predefined classname you can hook styling to. Nice.

File: config/yo/generator-component/app/templates/Component.css [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+@import "src/css/variables/colors.css";
+
+.<%= className %> {
+  padding: 8px;
```
Comment (rese) [inline config/yo/generator-component/app/templates/Component.css]: What's the default padding for?

Comment (unindented) [reply config/yo/generator-component/app/templates/Component.css]: Nothing, it was just random CSS contents. Hope people don't take it as a requirement to have 8px padding...

**Pull request merged**